components/html-block/html-block.js

```diff
@@ -274,20 +293,14 @@ class HtmlBlock extends RtlMixin(LitElement) {
 	}
 
 	_stamp(slot) {
-		const stampHTML = async nodes => {
-			if (nodes && nodes.length > 0) {
-
-				let temp = document.createElement('div');
-				temp.style.display = 'none';
-				nodes.forEach(node => temp.appendChild(node.cloneNode(true)));
+		const renderContainer = this.shadowRoot.querySelector('.d2l-html-block-rendered');
 
-				this._renderContainer.appendChild(temp);
-				temp = await this._processRenderers(temp);
-				this._renderContainer.innerHTML = temp.innerHTML;
+		const stampHTML = async nodes => {
+			renderContainer.innerHTML = '';
```

Comment:
Wipe the render container before we start stamping. We can't assume that whatever was rendered here before stamping is still valid. Prism and MathJax both effectively need us to start from a clean slate, for example.